packages/arb-util/broadcaster/clientmanager.go

```diff
@@ -271,25 +263,22 @@ func (cm *ClientManager) startWriter(ctx context.Context) {
 			case <-ctx.Done():
 				return
 			case data := <-cm.out:
-				cm.mu.RLock()
-				// Copy list so data can be written to each client without lock held
-				clientList := make([]*ClientConnection, len(cm.clientPtrMap))
-				var i uint64
+				clientDeleteList := make([]*ClientConnection, 0, len(cm.clientPtrMap))
+				cm.mu.Lock()
```

Comment:
Not sure if this will be an issue, but if we're getting a lot of messages this thread could try to hold the mutex almost all the time. Probably should leave as-is for now, but something to look out for

Reply:
sounds good, can send up to X number of messages if ready to send each time mutex is locked